numpy/distutils/tests/test_exec_command.py

```diff
@@ -90,3 +92,128 @@ def test_exec_command_stderr():
+        self.assertEqual(s, 0)
+        self.assertEqual(o, 'Tere')
+
+        s, o = exec_command.exec_command('echo "$AAA"', **kws)
+        self.assertEqual(s, 0)
+        self.assertEqual(o, '')
+
+        if 'BBB' not in os.environ:
+            os.environ['BBB'] = 'Hi'
+            s, o = exec_command.exec_command('echo "$BBB"', **kws)
+            self.assertEqual(s, 0)
+            self.assertEqual(o, 'Hi')
+
+            s, o = exec_command.exec_command('echo "$BBB"', BBB='Hey', **kws)
+            self.assertEqual(s, 0)
+            self.assertEqual(o, 'Hey')
+
+            s, o = exec_command.exec_command('echo "$BBB"', **kws)
+            self.assertEqual(s, 0)
+            self.assertEqual(o, 'Hi')
+
+            del os.environ['BBB']
+
+            s, o = exec_command.exec_command('echo "$BBB"', **kws)
+            self.assertEqual(s, 0)
+            self.assertEqual(o, '')
+
+
+        s, o = exec_command.exec_command('this_is_not_a_command', **kws)
+        self.assertNotEqual(s, 0)
+        self.assertNotEqual(o, '')
+
+        s, o = exec_command.exec_command('echo path=$PATH', **kws)
+        self.assertEqual(s, 0)
+        self.assertNotEqual(o, '')
+
+        s, o = exec_command.exec_command(
+             '"%s" -c "import sys,os;sys.stderr.write(os.name)"' %
+             self.pyexe, **kws)
+        self.assertEqual(s, 0)
+        self.assertEqual(o, 'posix')
+
+    def check_basic(self, *kws):
+        s, o = exec_command.exec_command(
+                     '"%s" -c "raise \'Ignore me.\'"' % self.pyexe, **kws)
+        self.assertNotEqual(s, 0)
+        self.assertNotEqual(o, '')
+
+        s, o = exec_command.exec_command(
+             '"%s" -c "import sys;sys.stderr.write(\'0\');'
+             'sys.stderr.write(\'1\');sys.stderr.write(\'2\')"' %
+             self.pyexe, **kws)
+        self.assertEqual(s, 0)
+        self.assertEqual(o, '012')
+
+        s, o = exec_command.exec_command(
+                 '"%s" -c "import sys;sys.exit(15)"' % self.pyexe, **kws)
+        self.assertEqual(s, 15)
+        self.assertEqual(o, '')
+
+        s, o = exec_command.exec_command(
+                     '"%s" -c "print(\'Heipa\'")' % self.pyexe, **kws)
+        self.assertEqual(s, 0)
+        self.assertEqual(o, 'Heipa')
+
+    def check_execute_in(self, **kws):
+        with tempdir() as tmpdir:
+            fn = "file"
+            tmpfile = os.path.join(tmpdir, fn)
+            f = open(tmpfile, 'w')
+            f.write('Hello')
+            f.close()
+
+            s, o = exec_command.exec_command(
+                 '"%s" -c "f = open(\'%s\', \'r\'); f.close()"' %
+                 (self.pyexe, fn), **kws)
+            self.assertNotEqual(s, 0)
+            self.assertNotEqual(o, '')
+            s, o = exec_command.exec_command(
+                     '"%s" -c "f = open(\'%s\', \'r\'); print(f.read()); '
+                     'f.close()"' % (self.pyexe, fn), execute_in=tmpdir, **kws)
```

Comment:
I think this is worse. because %s might contain literal backslashes (windows, for instance). Can you read from `sys.argv` instead, so you only need to shell-escape it?